### ISSUE TITLE
Maintain chain information from prediction functions

### ISSUE THREAD

**Issue by n-kall** [paul-buerkner/brms#1534]
Currently `posterior_predict` and similar functions return an array of draws (not a `draws_array` from `posterior`), which is S x N dimensions. I would like to combine this array of predictions with the posterior draws (in `draws` format), but as this array does not have any record of the chains, `posterior::bind_draws` does not work ("chain_ids do not match").

Example

```{r}
fit <- brm(yield ~ N * P * K, npk)
posterior::bind_draws(as_draws(fit), as_draws(posterior_predict(fit)))
#> Error: 'chain_ids' of bound objects do not match.
```

I'd like to avoid having to merge the chains of the posterior draws, so if it would be possible for the prediction functions from brms to return chain information (e.g. using the `draws_array` format), that would be very useful.

**Comment by paul-buerkner** [comment]
Agreed. However, I cannot just change the output format of such a frequently used function. This has to be part of brms 3.0. I am definitely planning on outputting posterior draws formats. Which of them it will become, I am not yet sure.

**Comment by avehtari** [comment]
Would it be possible to add an option to choose the output format, and keep the current format as the default?
